### PR TITLE
[windows] Check if Windows Defender is installed

### PR DESCRIPTION
**Pull request by saifshaikh48**
This PR fixes an issue where containerd configuration fails on instances
that do not have Windows Defender installed. WCMO now checks for Windows
Defender to be present before it creates an antivirus exclusion for containerd.

### CODE REVIEW

**Comment by mansikulkarni96** [comment]
Thanks for working on the quick fix @saifshaikh48 
/lgtm

**Comment by mansikulkarni96** [comment]
fixes #1107

File: pkg/windows/windows.go [hunk 1]
```diff
@@ -446,14 +446,17 @@ func (vm *windows) Configure() error {
 // configureContainerd configures the Windows defender exclusion and starts the
 // Windows containerd service
 func (vm *windows) configureContainerd() error {
-	// set Windows defender exclusions for containerd
-	setExclusionCmd := "Add-MpPreference -ExclusionProcess " + containerdPath
-	out, err := vm.Run(setExclusionCmd, true)
+	// if Windows Defender is installed on the instance, create an exlcusion for containerd
```
Comment (sebsoto) [inline pkg/windows/windows.go:449]: exlcusion -> exclusion

File: pkg/windows/windows.go [hunk 2]
```diff
@@ -1037,6 +1040,27 @@ func (vm *windows) isContainersFeatureEnabled() (bool, error) {
...
+	}
+	return strings.TrimSpace(out) == "True", nil
+}
+
+// createWindowsDefenderExclusion sets a Windows Defender exclusion for the given file
+func (vm *windows) createWindowsDefenderExclusion(pathToFile string) error {
+	command := "Add-MpPreference -ExclusionProcess " + pathToFile
+	out, err := vm.Run(command, true)
+	if err != nil {
+		vm.log.V(1).Info("setting Windows defender exclusion failed", "command", command, "output", out)
```
Comment (sebsoto) [inline pkg/windows/windows.go:1058]: is this log necessary? Its duplicate information that will be present in the error being returned.
If the `out` is valuable info, wouldnt it make sense to add it to the error message?

**Review by sebsoto — CHANGES_REQUESTED** [review]
Thanks for jumping on this quickly

File: pkg/windows/windows.go (hunk 2 above)
Comment (saifshaikh48) [reply pkg/windows/windows.go:1058]: You're right, I will fix this

**Comment by saifshaikh48** [comment]
@jfrancoa @rrasouli 

Want to bring this PR to the attention of QE

**Comment by jfrancoa** [comment]
> @jfrancoa @rrasouli
> 
> Want to bring this PR to the attention of QE

Should we create a new vSphere golden image without Windows Defender to verify this? Or could we use, for example the BYOH workflow (there we also have control on what the instance has enabled before WMCO adds it into the cluster). So, changing the user_data and making sure Windows Defender isn't enabled, should still allow us creating workloads on that BYOH node, right? However, if you consider we need to create a new vSphere golden image we can do it too.

**Comment by rrasouli** [comment]
@saifshaikh48 What is the outcome of NOT having Windows Defender, ERROR, INFO failed test?

**Comment by philipp1992** [comment]
do we need to install windows defender for the WMCO to work? we have a BYOH windows VM without defender and the WMCO throws the following error:

`{"level":"error","ts":1666266078.4728394,"msg":"Reconciler error","controller":"certificatesigningrequest","controllerGroup":"certificates.k8s.io","controllerKind":"CertificateSigningRequest","machine":{"name":"chmuw-default-windows-sq2nv","namespace":"openshift-machine-api"},"namespace":"openshift-machine-api","name":"chmuw-default-windows-sq2nv","reconcileID":"30aa1ad4-4645-44ce-b107-5dcd044fe0f1","error":"unable to configure instance 42245dad-2e8c-be76-eb07-aa192be0911e: configuring the Windows VM failed: error configuring containerd: setting Windows defender process exclusion failed with output: Add-MpPreference : Invalid class \r\nAt line:1 char:1\r\n+ Add-MpPreference -ExclusionProcess C:\\k\\containerd\\containerd.exe\r\n+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : MetadataError: (MSFT_MpPreference:root\\Microsoft\\...FT_MpPreference) [Add-MpPreference], \r\n    CimException\r\n    + FullyQualifiedErrorId : HRESULT 0x80041010,Add-MpPreference\r..`

We are running OCP 4.11.8

We do NOT want to use Defender

**Comment by sebsoto** [comment]
@philipp1992 Please see https://github.com/openshift/windows-machine-config-operator/issues/1240

**Comment by saifshaikh48** [comment]
> What is the outcome of NOT having Windows Defender, ERROR, INFO failed test?

The outcome should be that the containerd service is properly configured/started and the Windows instance is successfully configured as a node without any errors.

> Should we create a new vSphere golden image without Windows Defender to verify this? Or could we use, for example the BYOH workflow (there we also have control on what the instance has enabled before WMCO adds it into the cluster). So, changing the user_data and making sure Windows Defender isn't enabled, should still allow us creating workloads on that BYOH node, right? However, if you consider we need to create a new vSphere golden image we can do it too.

Either method should work fine. The main thing here is to test have tests verifying functionality when Windows Defender is not enabled and when it is not

**Pull request merged**